app/src/main/java/ar/uba/fi/mercadolibre/activity/UserPurchasesActivity.java

```diff
@@ -0,0 +1,61 @@
+package ar.uba.fi.mercadolibre.activity;
+
+import android.os.Bundle;
+import android.os.PersistableBundle;
+import android.support.annotation.Nullable;
+import android.util.Log;
+import android.view.View;
+import android.widget.ListView;
+import android.widget.TextView;
+
+import java.util.List;
+
+import ar.uba.fi.mercadolibre.R;
+import ar.uba.fi.mercadolibre.adapter.PurchasesAdapter;
+import ar.uba.fi.mercadolibre.controller.APIResponse;
+import ar.uba.fi.mercadolibre.controller.ControllerFactory;
+import ar.uba.fi.mercadolibre.model.Purchase;
+import retrofit2.Call;
+import retrofit2.Callback;
+import retrofit2.Response;
+
+public class UserPurchasesActivity extends BaseActivity {
+    @Override
+    public int identifierForDrawer() {
+        return MY_PURCHASES_IDENTIFIER;
+    }
+
+    @Override
+    protected void onCreate(Bundle savedInstanceState) {
+        super.onCreate(savedInstanceState);
+        setContentView(R.layout.activity_user_purchases);
+        init();
+    }
+
+    private void init() {
+        ControllerFactory.getPurchaseController().getUserPurchases().enqueue(new Callback<APIResponse<List<Purchase>>>() {
+            @Override
+            public void onResponse(Call<APIResponse<List<Purchase>>> call, Response<APIResponse<List<Purchase>>> response) {
+                List<Purchase> data = getData(response);
```

Comment:
/**
     * Tries to return data wrapped by APIResponse.
     * If it can't, it will call onGetDataFailure and return null.
     *
     * @return Data wrapped by APIResponse or null if there was an error
     */

se que la solucion de 'guarda que getdata si falla da null' no es muy bella, pero bueno jajaja el punto es que abajo de un getData deberia haber un if data == null return

Reply:
que raro que no te chillo androidstudio con que data podia ser null... (o lo ignoraste jajaj)